app/api/api_v1/routers/admin.py

```diff
@@ -71,10 +71,29 @@ async def update_document(
     # Note this code relies on the fields being the same as the db column names
     num_changed = db.execute(
         update(PhysicalDocument)
-        .values(meta_data.dict())
+        .values(meta_data.physical_doc_keys_json())
         .where(PhysicalDocument.id == physical_document.id)
     ).rowcount
 
+    # Update the languages
+    if meta_data.languages is not None:
+        _LOGGER.info(
+            "Adding meta_data object languages to the database.",
+            extra={"props": {'meta_data_languages': meta_data.languages}}
+        )
+
+        db.query(PhysicalDocumentLanguage).filter(PhysicalDocumentLanguage.document_id == physical_document.id).delete()
+        db.flush()
+
+        for language in meta_data.languages:
+            lang = db.query(Language).filter(Language.language_code == language).one_or_none()
+            if lang is not None:
+                physical_document_language = PhysicalDocumentLanguage(
+                    language_id=lang.id, document_id=physical_document.id
+                )
+                db.add(physical_document_language)
```

Comment:
This will throw errors if a language association already exists

Reply:
I'm not sure I follow why? 
On line 85 we remove all PhysicalDocumentLanguage associations from the database and then re-populate the associations from the languages declared in the request object?

Reply:
I missed the deletion... but I can't say I'm super comfortable with the approach. If we delete all the language associations, then fail during updates we lose information. Is it possible to calculate the difference, and therefore the ones we need to add?